cmd/satellite/main.go

```diff
@@ -226,6 +243,37 @@ func cmdDiag(cmd *cobra.Command, args []string) (err error) {
 	return err
 }
 
+func cmdQDiag(cmd *cobra.Command, args []string) (err error) {
+	// open the psql db
+	dbpath := qdiagCfg.BasePath
+
+	redisQ, err := redis.NewQueueFrom(dbpath)
+	if err != nil {
+		return err
+	}
+
+	queue := queue.NewQueue(redisQ)
+	repairer := repairer.NewRepairer(queue, nil, qdiagCfg.Interval, qdiagCfg.MaxRepair)
+	list, err := repairer.Queue.Peekqueue()
+	if err != nil {
+		return err
+	}
+
+	// initialize the table header (fields)
+	const padding = 3
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, padding, ' ', tabwriter.AlignRight|tabwriter.Debug)
+	fmt.Fprintln(w, "Path\tLost Pieces\t")
+
+	// populate the row fields
+	for _, v := range list {
+		fmt.Fprint(w, v.GetPath(), "\t", v.GetLostPieces(), "\t")
+	}
+
+	// display the data
+	err = w.Flush()
```

Comment:
You can move `w.Flush()` directly to `return` line

Reply:
Done